packages/application-shell/src/components/quick-access/index.js

```diff
@@ -61,6 +61,21 @@ export default class QuickAccessTrigger extends React.Component {
   state = {
     isVisible: false,
     hasError: false,
+    // We store the information of whether a project is indexed by pim-indexer,
+    // to avoid having to refetch that information every time Quick Access is
+    // opened. We can't move the information to the quick-access.js component
+    // as that component unmounts and would lose its state.
+    //
+    // We need to know whether a project is indexed by pim-indexer to know
+    // whether we should query pim-search or whether we can skip that request.
+    //
+    // We don't need to update this information when the project key changes,
+    // as changing a project always results in a full page reload anyways.
+    //
+    // undefined: we did not check yet
+    // true: the project is indexed by pim-indexer
+    // false: the project is not indexed by pim-indexer
+    isProjectPimIndexed: undefined,
```

Comment:
While `undefined` semantics of booleans in JS is kind of legit, I personally don't like this three-state-booleans, for exactly this reason: you have to add comments all around this places, so that people reading will take `undefined` into consideration.

Reply:
Yeah that bothered me as well while writing that. I can create an enum for it instead to make it clearer:

```
const pimIndexerStates = 
  UNCHECKED: 'UNCHECKED',
  INDEXED: 'INDEXED',
  NOT_INDEXED: 'NOT_INDEXED',
}
```